debezium-embedded/src/main/java/io/debezium/embedded/DefaultOffsetManager.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.embedded;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.json.JsonConverterConfig;
+import org.apache.kafka.connect.runtime.WorkerConfig;
+import org.apache.kafka.connect.storage.Converter;
+import org.apache.kafka.connect.storage.OffsetBackingStore;
+import org.apache.kafka.connect.storage.OffsetStorageReader;
+import org.apache.kafka.connect.storage.OffsetStorageReaderImpl;
+import org.apache.kafka.connect.storage.OffsetStorageWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.config.Instantiator;
+
+/**
+ * Default implementation of {@link OffsetManager} that uses Kafka's OffsetStorageWriter to commit offsets.
+ * This class is meant to be used in a thread confined manner and is not thread safe.
+ */
+public class DefaultOffsetManager implements OffsetManager {
```

Comment:
I'm inclined that we rename this from `DefaultOffsetManager` to `KafkaOffsetManager`, following a similar naming that we use for storage-based implementations as well as the `OffsetBackingStore` implementations.  

Also, annotate this class with `@NotThreadSafe` if its expressly not thread-safe.